### ISSUE TITLE
REPL output is truncated

### ISSUE THREAD

**Issue by failable** [zed-industries/zed#16934]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Beginning of the REPL output is truncated. The issue remains if I make the Zed window wider and re-evaluate.

Left: Kitty Terminal Right: Zed
<img width="1920" alt="Screenshot 2024-08-27 at 14 40 54" src="https://github.com/user-attachments/assets/90fd93de-50ca-4869-b09d-c23f9aeff449">

### Environment

Zed: v0.150.3 (Zed Preview)
OS: macOS 14.6.1
Memory: 32 GiB
Architecture: x86_64

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by pantheraleo-7** [comment]
`stderr` almost always get truncated

**Comment by pantheraleo-7** [comment]
yes (I'm on preview), but a small error (so no open button) needs to be copied and then pasted in an empty buffer which is quite tiresome 😃. This truncation problem needs to be fixed.

If the repl output area is a terminal, can it not be sized according to the pane? (I haven't seen the code though)

**Comment by failable** [comment]
Not fixed yet.

**Comment by Dimentium** [comment]
still reproducible with:
- zed 0.178.5 
- macOS 15.2
- ipython 9.0.2

<img width="1544" alt="Image" src="https://github.com/user-attachments/assets/439e5d98-8a5b-4ee5-83fa-09031ad8c617" />

**Comment by failable** [comment]
Not fixed yet
